src/base/QXmppJingleData.h

```diff
@@ -657,6 +658,73 @@ class QXMPP_EXPORT QXmppJingleMessageInitiationElement
     QSharedDataPointer<QXmppJingleMessageInitiationElementPrivate> d;
 };
 
+class QXMPP_EXPORT QXmppCallInviteElement
+{
+public:
+    enum class Type {
+        None,
+        Invite,
+        Retract,
+        Accept,
+        Reject,
+        Left
+    };
+
+    struct Jingle
+    {
+        QString sid;
+        std::optional<QString> jid;
+
+        bool operator==(const Jingle &other) const { return other.sid == sid && other.jid == jid; }
+
+        void parse(const QDomElement &element);
+        void toXml(QXmlStreamWriter *writer) const;
+    };
+
+    struct External
+    {
+        External(QString uri) : uri(uri) {};
+        QString uri;
+
+        bool operator==(const External &other) const { return other.uri == uri; }
+
+        void toXml(QXmlStreamWriter *writer) const;
+    };
+
+    QXmppCallInviteElement();
+    QXMPP_PRIVATE_DECLARE_RULE_OF_SIX(QXmppCallInviteElement)
+
+    Type type() const;
+    void setType(Type type);
+
+    QString id() const;
+    void setId(const QString &id);
+
+    bool audio() const;
+    void setAudio(bool audio);
+
+    bool video() const;
+    void setVideo(bool video);
+
+    std::optional<Jingle> jingle() const;
+    void setJingle(std::optional<Jingle> jingle);
+
+    std::optional<QVector<External>> external() const;
+    void setExternal(std::optional<QVector<External>> external);
+
+    /// \cond
+    void parse(const QDomElement &element);
+    void toXml(QXmlStreamWriter *writer) const;
+    /// \endcond
+
+    static bool isCallInviteElement(const QDomElement &);
+    static QString callInviteElementTypeToString(Type type);
+    static std::optional<Type> stringToCallInviteElementType(const QString &typeStr);
```

Comment:
The type parse functions should probably rather be private.

Reply:
It's just that if they're `public`, they are part of the public QXmpp API and need to stay compatible. APIs that are not needed by the users of QXmpp shouldn't be public, so we don't need to keep the compaibility of them.

Better make it `private:` or put it into `namespace QXmpp::Private {}`, if they are not needed by users of QXmpp.